wordpress-mobile/WordPress-iOS
# Comment Moderation Bar: update comment when Spam button tapped.

Ref: #17200 

When the `Spam` button is tapped on the moderation bar, the comment status is now updated.
- A success message appears when successful.
- When unsuccessful, a failure message appears and the button states are reset.

To test:
- Enable the `newCommentDetail` feature.
- Go to My Site > Comments.
- Select a comment that is _not_ spam.
- Select `Spam` on the moderation bar.
- Verify a success message is displayed.
- Go back to the comments list, and select the Spam filter.
- Verify the comment appears in the list.

https://user-images.githubusercontent.com/1816888/136435306-67a6a172-1ccd-4525-b1f5-f34ac7b894c0.mp4

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete.

PR submission checklist:
